rmgpy/data/kinetics/common.py

```diff
@@ -251,16 +255,18 @@ def independent_ids():
     if not independent_ids():
         logging.debug('identical atom ids found between species. regenerating')
         for species in input_species:
-            mol = species.molecule[0]
-            mol.assignAtomIDs()
-            species.molecule = [mol]
-            # Remake resonance structures with new labels
-            if resonance:
-                species.generate_resonance_structures(keepIsomorphic=True)
+            if all([mol.reactive for mol in species.molecule]):
```

Comment:
What are potential cases where there might be an unreactive molecules when calling this method? Independent atom IDs are pretty important for degeneracy, so depending on the situation, a more comprehensive fix might be better.

Reply:
Molecules with a `reactive=False` flag get generated, but are not allowed to further react in families. However, we do calculate degeneracy for these reactions. My concern here was a line below that resets the molecule list (`species.molecule = [mol]`), removing all unreactive molecules. Even if we re-generate resonance structures, we won't regenerate these unreactive molecule back. I modified this that condition so that unreactive molecules are appended if needed.